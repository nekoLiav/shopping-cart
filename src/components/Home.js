const Home = () => {
  return (
    <div className='home'>
      <div className='home-content'>
        <h1>Home</h1>
      </div>
    </div>
  );
};

export default Home;
