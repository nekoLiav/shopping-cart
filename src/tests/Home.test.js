import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';
import userEvent from '@testing-library/user-event';

describe('Shop component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders correct heading', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(getByRole('heading', { name: 'Home' }));
  });
  it('navigates to shop when Go To Shop button is clicked', async () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const user = userEvent.setup();
    await user.click(screen.getByText('Go To Shop'));
    expect(screen.getByRole('heading', { name: 'Home' }));
  });
});
