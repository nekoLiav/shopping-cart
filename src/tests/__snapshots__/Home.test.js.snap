// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component matches snapshot 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="page-header-text"
    >
      Home
    </h1>
    <div
      class="home-content"
    >
      <p
        class="home-super-important-disclaimer"
      >
        This isn't a real shop, but it's still sorta neat.
      </p>
      <a
        href="/shop"
      >
        Go To Shop
      </a>
    </div>
  </div>
</div>
`;
